src/dlgTriggerEditor.cpp

```diff
@@ -6957,25 +6957,28 @@ void dlgTriggerEditor::slot_script_main_area_delete_handler()
 
 void dlgTriggerEditor::slot_script_main_area_add_handler()
 {
+    auto createItemList = [](auto * mpScriptsMainArea) {
+        auto pItem = new QListWidgetItem;
+        pItem->setText(mpScriptsMainArea->lineEdit_script_event_handler_entry->text());
+        mpScriptsMainArea->listWidget_script_registered_event_handlers->addItem(pItem);
+        return pItem;
+    };
+
     mpScriptsMainArea->trimEventHandlerName();
     if (mIsScriptsMainAreaEditHandler) {
         if (!mpScriptsMainAreaEditHandlerItem) {
             mIsScriptsMainAreaEditHandler = false;
-            goto LAZY;
+            auto pItem = createItemList(mpScriptsMainArea);
             return;
```

Comment:
Ah, this line is, of course, unreachable.

Reply:
It's reachable, it runs after the lambda, no?

What I'm confused about is why is it here - won't the `lineedit` with the event handler name not be cleared then?

Reply:
`lineEdit_script_event_handler_entry->text()` ?

Reply:
btw there is already `mpScriptsMainArea->lineEdit_script_event_handler_entry->clear();` at near end of the function.

Reply:
ahha. so it return without `clear()` it. mhmm

Reply:
applied.